tools/custom-stats-qc/1.0.0/custom-stats-qc__1.0.0.cwl

```diff
@@ -0,0 +1,154 @@
+          echo \$JSON_STRING >> "$(inputs.output_json_filename).json"
+
+          # Extract pct_discordant_read_pairs
+          pct_discordant_reads=\$(grep -h "SN	percentage of properly paired reads (%):" "$(inputs.output_samtools_stats.path)" | cut -f3)
+          pdr=\$(awk -vn1="\$pct_discordant_reads" 'BEGIN { print (100 - n1) }')
+          JSON_STRING=\$( jq -n \\
+                            --arg des "pct discordant reads" \\
+                            --arg src "SN	percentage of properly paired reads (%)" \\
+                            --arg val "$pct_discordant_reads" \\
+                            '{pct_discordant_reads: {description: $des, source: $src, value: $val}}' )
+          echo \$JSON_STRING >> "$(inputs.output_json_filename).json"
+
+          # Extract mean_insert_size
+          mean_insert_size=\$(grep -h "SN	insert size average:" "$(inputs.output_samtools_stats.path)" | cut -f3)
+          JSON_STRING=\$( jq -n \\
+                            --arg des "mean insert size" \\
+                            --arg src "SN	insert size average" \\
+                            --arg val "$mean_insert_size" \\
+                            '{mean_insert_size: {description: $des, source: $src, value: $val}}' )
+          echo \$JSON_STRING >> "$(inputs.output_json_filename).json"
+
+          # Extract Insert_size_SD
+          insert_size_sd=\$(grep -h "SN	insert size standard deviation:" "$(inputs.output_samtools_stats.path)" | cut -f3)
+          JSON_STRING=\$( jq -n \\
+                            --arg des "insert size sd" \\
+                            --arg src "SN	insert size standard deviation" \\
+                            --arg val "$insert_size_sd" \\
+                            '{insert_size_sd: {description: $des, source: $src, value: $val}}' )
+          echo \$JSON_STRING >> "$(inputs.output_json_filename).json"
+
+          # Extract pct_mapped_reads where pct_mapped_reads = [(Mapped Reads - Reads MQ0) / Total Reads]
+          # RM = Mapped Reads
+          # R0 = Reads MQ0
+          # RT = Total Reads
+          RM=\$(grep -h "SN	reads mapped:" "$(inputs.output_samtools_stats.path)" | cut -f3)
+          R0=\$(grep -h "SN	reads MQ0:" "$(inputs.output_samtools_stats.path)" | cut -f3)
+          RT=\$(grep -h "SN	raw total sequences:" "$(inputs.output_samtools_stats.path)" | cut -f3)
+          pct_mapped_reads=\$(( 100 * ( $RM - $R0) / $RT ))
+          JSON_STRING=\$( jq -n \\
+                            --arg des "pct mapped reads" \\
+                            --arg src "pct_mapped_reads = [(Mapped Reads - Reads MQ0) / Total Reads]" \\
+                            --arg val "$pct_mapped_reads" \\
+                            '{pct_mapped_reads: {description: $des, source: $src, value: $val}}' )
+          echo \$JSON_STRING >> "$(inputs.output_json_filename).json"
+
+          jq '.' "$(inputs.output_json_filename).json" >> "$(inputs.output_json_filename)_$(inputs.sample_id).json"
```

Comment:
This doesn't currently output a valid json file.  Copy and paste it into a IDE and it will come up with a lot of red lines.  See below:
![image](https://user-images.githubusercontent.com/8197659/149367281-0c46671c-1435-40ac-9eb2-7efb9a078c5c.png)


A workaround is substituting `'.'` for `[inputs] | add` which will put everything under one dictionary.

i.e
```
{
  "average_base_quality": {
    "description": "average base quality",
    "source": "SN\taverage quality",
    "value": "29.5"
  },
  "yield_mapped_bp": {
    "description": "yield mapped bp",
    "source": "SN\tbases mapped (cigar)",
    "value": "92677929585"
  },
  "pct_autosomes_20x": {
    "description": "pct autosomes 20x",
    "source": "SN\tpercentage of target genome with coverage > 20",
    "value": "92.31"
  },
  ...
}  
```